src/FieldWarning/Assets/UI/Ingame/UnitLabel/OverlayFactory.cs

```diff
@@ -14,36 +14,27 @@
 using PFW.Units;
 using UnityEngine;
 
-public class OverlayFactory
+namespace PFW.UI.Ingame.UnitLabel
 {
-    private static OverlayFactory instance; // Needed
-
-    private OverlayFactory()
-    { 
-        instance = this;
-    }
-
-    public static OverlayFactory Instance()
+    public sealed class OverlayFactory
     {
-        if (instance == null)
+        private OverlayFactory()
         {
-            instance = new OverlayFactory();
         }
 
-        return instance;
-    }
-
-    public WaypointOverlayBehavior CreateWaypointOverlay(PlatoonBehaviour pb)
-    {
-        var overlayPrefab = Resources.Load<GameObject>("WaypointOverlay");
+        public static OverlayFactory Instance { get; } = new OverlayFactory();
 
-        var waypointOverlayBehavior = 
-            Object.Instantiate(overlayPrefab, Vector3.zero, Quaternion.identity).
-                GetComponent<WaypointOverlayBehavior>();
+        public WaypointOverlayBehavior CreateWaypointOverlay(PlatoonBehaviour pb)
+        {
+            var overlayPrefab = Resources.Load<GameObject>("WaypointOverlay");
 
-        waypointOverlayBehavior.Initialize(pb);
+            var waypointOverlayBehavior = Object.Instantiate(
+                    overlayPrefab, Vector3.zero, Quaternion.identity)
+                .GetComponent<WaypointOverlayBehavior>();
```

Comment:
The indentation doesn't look right here although I understand the rules that led to it

Reply:
I will move Object.Instantiate on the next line to its parameters